src/Symfony/Component/Cache/SimpleCache.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\SimpleCache\CacheInterface;
+use Psr\SimpleCache\CounterInterface;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class SimpleCache implements CacheInterface, CounterInterface
+{
+    private $pool;
+    private $createCacheItem;
+
+    public function __construct(CacheItemPoolInterface $pool)
+    {
+        $this->pool = $pool;
+
+        if ($pool instanceof Adapter\AdapterInterface) {
+            $this->createCacheItem = \Closure::bind(
+                function ($key, $value) {
+                    CacheItem::validateKey($key);
+                    $item = new CacheItem();
+                    $item->key = $key;
+                    $item->value = $value;
+
+                    return $item;
+                },
+                null,
+                CacheItem::class
+            );
+        } else {
+            $this->createCacheItem = function ($key, $value) {
+                return $this->pool->getItem($key)->set($value);
+            };
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($key, $default = null)
+    {
+        $item = $this->pool->getItem($key);
+
+        return $item->isHit() ? $item->get() : $default;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function set($key, $value, $ttl = null)
+    {
+        $f = $this->createCacheItem;
+        $item = $f($key, $value);
+        if (null !== $ttl) {
+            $item->expiresAfter($ttl);
+        }
+
+        return $this->pool->save($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete($key)
+    {
+        return $this->pool->deleteItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        return $this->pool->clear();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMultiple($keys)
+    {
+        foreach ($this->pool->getItems($keys) as $key => $item) {
+            yield $key => $item->get();
```

Comment:
Not really on purpose. Please email on fig ML to keep everything centralized there because otherwise stuff will get lost.. Not sure if I like generators here though, if you get multiple keys you most likely want to have them in an array where you access every result by key, and not have to loop over it.. But it's also kinda easy to wrap iterator_to_array so not huge deal either way.